### PR TITLE
Fixes shift click selection after programmatic selection in most cases, Issue #2469

### PR DESCRIPTION
**Pull request by RRomeroJr**
Proposed fix for issue #2469
Original thread:
https://forums.ankiweb.net/t/bug-arrow-keys-in-preview-break-shift-click-selection-in-card-browser/29282

Some things I noticed while looking into this.

- Changing the selection programmatically followed by shift-click selecting is what causes the issue. 
This would happen when:
    - `<` /  `>` buttons were clicked in the preview window
    - `Ctrl + N` or `Ctrl+ P` table shortcuts were used in the browser window
    - `Home` or `End` table shortcuts were used in the browser window
   
   Then..
   - `Shift + Left Click` selecting in the table view
- Some quick googling showed that this was a common issue with Qt, and I reproduced it in a small mockup program I made. I'm not sure if it is a bug with Qt itself, or if there is some better/ more proper way to select items in a TableView from code.

What I added feels like a work-around more than a proper fix, but, as far as I can tell, it fixes all the mentioned scenarios.

**However**, if the user holds `Shift` while clicking the `<` or `>`  buttons in the preview window the same bug reappears. I haven't been able to figure out why this still happens.

### CODE REVIEW

**Comment by RRomeroJr** [comment]
Sorry about making 2 pull requests. In hindsight I could've fixed the other one.

If this is acceptable, I'd like to ask one question. I'd like to try to make an automated test for this behavior, but I'm not sure were to start looking implement it. If possible, any advice on where to look would be appreciated.

**Comment by dae** [comment]
Thanks for looking into this! Just one minor request: please link to the issue in the comment, to make it a little easier to find the related discussion.

> What I added feels like a work-around more than a proper fix, but, as far as I can tell, it fixes all the mentioned scenarios.

Even if it's not perfect, if it's less bad than what we currently have, that's a win.

**Comment by RRomeroJr** [comment]
Added link up top 👍

File: qt/aqt/browser/table/table.py [hunk 1]
```diff
@@ -608,7 +608,9 @@ def _move_current(
                 direction,
                 self.browser.mw.app.keyboardModifiers(),
             )
-        self._view.selectionModel().setCurrentIndex(
+        # Setting current like this avoids a bug with shift-click selection
```
Comment (dae) [inline qt/aqt/browser/table/table.py:611]:
```suggestion
        # Setting current like this avoids a bug with shift-click selection
        # https://github.com/ankitects/anki/issues/2469
```

**Comment by dae** [comment]
Thank you!

**Pull request merged**